ocaml/testsuite/tests/templates/basic/bad_instance_arg_value_wrong_type.reference

```diff
@@ -2,4 +2,4 @@ File "bad_instance_arg_value_wrong_type.ml", line 2, characters 2-39:
 2 |   Category_utils(Category)(List_monoid) [@jane.non_erasable.instances]
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 Error: The module "List_monoid" was expected to satisfy the parameter "Category"
-       but "list_monoid.cmi" was compiled with -as-argument-for "Monoid".
+       but "list_monoid.cmi" was compiled to satisfy "Monoid".
```

Comment:
Sorry, I might have found a better sentence, how about:

> Error: The module "List_monoid" was expected as an argument for parameter "Category"
       but was compiled as an argument for parameter "Monoid".

(this is more consistent with our conventional wording "as an argument for parameter"

Reply:
Ah, yeah, that's even better

Reply:
Actually, I'll go one further and change both instances: “The module "List_monoid" is used as an argument for the parameter "Category" but "list_monoid.cmi" was compiled as an argument for "Monoid".”

Reply:
As discussed offline, we think this is even better:
> Error: The module "List_monoid" is used as an argument for parameter "Category"
but was compiled as an argument for parameter "Monoid".
Hint: `list_monoid.cmi` was compiled with `-as-argument-for Monoid`.